download-utilities.sh

```diff
@@ -800,6 +843,29 @@ if [[ "${yaml_file}" == "-" ]]; then
   cat > "${yaml_file}"
 fi
 
+# Handle YAML merging if requested
+if [ -n "${override_yaml_file:-}" ]; then
+  merged_yaml_file="$(mktemp)"
```

Comment:
```suggestion
  merged_yaml_file="${TMP_DIR}/merged.yaml"
```

TMP_DIR is created for temporary files and is automatically deleted when script exits.